code/game/gamemodes/umbrage/umbrage_objects.dm

```diff
@@ -0,0 +1,158 @@
+/obj/item/weapon/dark_bead
+	name = "dark bead"
+	desc = "A glowing black orb. It's fading fast."
+	icon_state = "dark_bead"
+	item_state = "ratvars_flame"
+	flags = NODROP
+	resistance_flags = FIRE_PROOF | LAVA_PROOF | UNACIDABLE | INDESTRUCTIBLE
+	w_class = 5
+	var/eating = 0 //If we're devouring someone's will
+	var/datum/action/innate/umbrage/devour_will/linked_ability //The ability that keeps data for us
+
+/obj/item/weapon/dark_bead/New()
+	..()
+	animate(src, alpha = 30, time = 30)
+	QDEL_IN(src, 30)
+
+/obj/item/weapon/dark_bead/Destroy(force)
+	if(isliving(loc) && !eating && !force)
+		loc << "<span class='warning'>You were too slow! [src] faded away...</span>"
+	if(!eating || force)
+		. = ..()
+	else
+		return QDEL_HINT_LETMELIVE
+
+/obj/item/weapon/dark_bead/attack(mob/living/carbon/L, mob/living/user)
+	if(!is_umbrage(user.mind) || eating || L == user) //no eating urself ;)))))))
+		return
+	var/datum/umbrage/U = linked_ability.linked_umbrage
+	if(!L.health)
+		user << "<span class='warning'>[L] is too weak to drain.</span>"
+		return
+	if(linked_ability.victims[L.real_name])
+		user << "<span class='warning'>[L] must be given time to recover from their last draining.</span>"
+		return
+	eating = 1
+	user.visible_message("<span class='warning'>[user] grabs [L] and leans in close...</span>", "<span class='velvet bold'>cera qo...</span><br>\
+	<span class='danger'>You begin siphoning [L]'s mental energy...</span>")
+	L << "<span class='userdanger'><i>AAAAAAAAAAAAAA-</i></span>"
+	L.Stun(3)
+	playsound(L, 'sound/magic/devour_will.ogg', 100, 0) //T A S T Y   S O U L S
+	if(!do_mob(user, L, 30))
+		user.Weaken(3)
+		L << "<span class='boldwarning'>All right. You're all right.</span>"
+		L.Weaken(3)
+		qdel(src)
+		return
+	user.visible_message("<span class='warning'>[user] gently lowers [L] to the ground...</span>", "<span class='velvet bold'>...aranupdejc</span><br>\
+	<span class='boldnotice'>You devour [L]'s will. Your psi has been fully restored.\n\
+	Additionally, you have gained one lucidity. Use it to purchase and upgrade abilities.</span><br>\
+	<span class='warning'>[L] is now severely weakened and will take some time to recover.</span>")
+	playsound(L, 'sound/magic/devour_will_victim.ogg', 50, 0)
+	U.psi = U.max_psi
+	U.lucidity++
+	U.lucidity_drained++
+	LAZYADD(U.drained_minds, L.mind)
+	linked_ability.victims[L.real_name] = L
+	L << "<span class='userdanger'>You suddenly feel... empty. Thoughts try to form, but flit away. You slip into a deep, deep slumber...</span>"
+	L << sound('sound/magic/devour_will_end.ogg', volume = 75)
+	L.Paralyse(30)
+	L.stuttering += 40
+	L.confused += 40
+	L.reagents.add_reagent("zombiepowder", 2) //Brief window of vulnerability to veiling
+	qdel(src, force = TRUE)
+	#warn Change this dark bead recovery timer - 2 minutes, maybe?
+	spawn(100) //I don't like to use a spawn here, but because of how it works I have to
```

Comment:
I wouldn't need to use a list in the ability if I used that, either. Not bad! I already have a very slightly hacky workaround in the next commit though, so I'll put that up for review and we can decide if it needs change.